### PR TITLE
feat: Reduce memory usage threshold limits

### PR DESCRIPTION
**Pull request by seeruyy**
Added threshold limits to control reduce memory usage:
 - reduceByteThresholdPerState
 - reduceByteThresholdAllStates

ref: LOG-19194

### CODE REVIEW

File: charts/edge/values.yaml [hunk 1]
```diff
@@ -78,3 +78,6 @@ pvcName: ""
 
 maxDiskBufferBytes: 0
 maxMemBufferEvents: 0
+
+reduceByteThresholdPerState: 0
```
Comment (aholmberg) [inline charts/edge/values.yaml:82]: I think the internal defaults are too low for almost any edge deployment. What would you think of setting these values higher (100MiB, 1GiB) and taking away the conditional templating in `statefulset.yaml`?

File: charts/edge/README.md [hunk 2]
```diff
@@ -29,20 +29,22 @@ helm install edge mezmo/edge \
...
+| service.sourcePorts.list     | array[]  | []              | Optional list of discrete ports to configure on the service
+| enableK8sLogs                | boolean  | false           | Whether or not to add ClusterRole and Volume access required for k8s logs source
+| pvcName                      | string   | ""              | Optional name for an existing PersistentVolumeClaim to use for disk buffering, aggregate persistence and k8s log checkpoints. Leaving undefined will default to ephemeral storage lasting the lifetime of the pod
+| maxDiskBufferBytes           | int      |                 | Optional max number of bytes to store in a disk-backed buffer. Leaving undefined will use default value of 268435488 (256MiB)
+| maxMemBufferEvents           | int      |                 | Optional max number of events to store in a memory buffer. Leaving undefined will use default value of 1000
+| reduceByteThresholdPerState  | int      |                 | Optional max number of bytes to store in a memory for any given reduce component. Leaving undefined will use default value of 102400 (100Kb)
```
Comment (aholmberg) [inline charts/edge/README.md:47]: Note there is a column for default, so we could remove that bit from the description

**Review by aholmberg — COMMENTED** [review]
Looks good. Couple of comments.

File: charts/edge/values.yaml (hunk 1 above)
Comment (seeruyy) [reply charts/edge/values.yaml:82]: Based on the [ticket](https://mezmo.atlassian.net/browse/LOG-19194) we're asked to set the limits to 10Mb and 100Mb, maybe lets set these values as default?

File: charts/edge/templates/statefulset.yaml [hunk 3]
```diff
@@ -100,6 +100,10 @@ spec:
           - name: MEZMO_MEM_BUFFER_MAX_EVENTS
             value: {{ .Values.maxMemBufferEvents }}
           {{- end }}
+          - name: REDUCE_BYTE_THRESHOLD_PER_STATE
+            value: {{ .Values.reduceByteThresholdPerState }}
```
Comment (aholmberg) [inline charts/edge/templates/statefulset.yaml:104]: Found out we need to quote these values. Also the two preexisting ones above them.

File: charts/edge/values.yaml (hunk 1 above)
Comment (aholmberg) [reply charts/edge/values.yaml]: That was for multitenant SaaS, right? I think we should default higher for this application.

Comment (seeruyy) [reply charts/edge/values.yaml]: Ok, we can do that

Comment (seeruyy) [reply charts/edge/values.yaml]: Increased to 100Mb and 1Gb

**Review by aholmberg — APPROVED** [review]
Looks great! Thanks for the quick turn.

**Pull request merged**